paho/message_ids.go

```diff
@@ -41,20 +41,29 @@ type CPContext struct {
 // to messages with a messageid
 type MIDs struct {
 	sync.Mutex
-	index map[uint16]*CPContext
+	index      map[uint16]*CPContext
+	lastIdUsed uint16
 }
 
 // Request is the library provided MIDService's implementation of
 // the required interface function()
 func (m *MIDs) Request(c *CPContext) (uint16, error) {
 	m.Lock()
 	defer m.Unlock()
+
+	if m.lastIdUsed < 65535 {
```

Comment:
I'm not sure why the guy was defining new constants while there are the math ones: https://golang.org/pkg/math/#pkg-constants

```go
const (
    MaxInt8   = 1<<7 - 1
    MinInt8   = -1 << 7
    MaxInt16  = 1<<15 - 1
    MinInt16  = -1 << 15
    MaxInt32  = 1<<31 - 1
    MinInt32  = -1 << 31
    MaxInt64  = 1<<63 - 1
    MinInt64  = -1 << 63
    MaxUint8  = 1<<8 - 1
    MaxUint16 = 1<<16 - 1
    MaxUint32 = 1<<32 - 1
    MaxUint64 = 1<<64 - 1
)
```

Reply:
Shouldn't this be like `0 <= lastIdUsed <= 65535`? :thinking: 
Looks like you're doing `1 <= lastIdUsed <= 65534`.

Reply:
@fracasula No it was correct 'cause I was incrementing below.